src/EnergyPlus/HeatingCoils.cc

```diff
@@ -3061,19 +3063,24 @@ namespace HeatingCoils {
             state.dataHeatingCoils->HeatingCoil(CoilNum).ParasiticFuelRate * ReportingConstant;
 
         std::string coilObjClassName;
-        {
-            auto const SELECT_CASE_var(state.dataHeatingCoils->HeatingCoil(CoilNum).HCoilType_Num);
-            if (SELECT_CASE_var == Coil_HeatingElectric) {
-                coilObjClassName = "Coil:Heating:Electric";
-            } else if (SELECT_CASE_var == Coil_HeatingElectric_MultiStage) {
-                coilObjClassName = "Coil:Heating:Electric:MultiStage";
-            } else if (SELECT_CASE_var == Coil_HeatingGasOrOtherFuel) {
-                coilObjClassName = "Coil:Heating:Fuel";
-            } else if (SELECT_CASE_var == Coil_HeatingGas_MultiStage) {
-                coilObjClassName = "Coil:Heating:Gas:MultiStage";
-            } else if (SELECT_CASE_var == Coil_HeatingDesuperheater) {
-                coilObjClassName = "Coil:Heating:Desuperheater";
-            }
+        switch (state.dataHeatingCoils->HeatingCoil(CoilNum).HCoilType_Num) {
```

Comment:
📌  Should be able to get these from the `constexpr std::array<std::string_view>` once these are converted to `enum class`.